spec/pQueue.js

```diff
@@ -0,0 +1,84 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import pQueue from '../src/pQueue'
+
+chai.use(chaiChange)
+
+describe('pQueue', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(pQueue).to.be.a('function')
+  })
+
+  context('enqueue()', () => {
+    it('adds an element with priority to the back of the queue.', () => {
+      const aQueue = new pQueue()
+
+      aQueue.enqueue('dude', 90)
+      aQueue.enqueue('cool', 20)
+      aQueue.enqueue('turd', 50)
+      expect(aQueue.enqueue('foo',70)).to.eql([{name: 'dude', priority: 90}, {name: 'cool',  priority: 20}, {name: 'turd', priority: 50}, {name: 'foo', priority: 70}])
+    })
+  })
+
+  // context('front()', () => {
+  //   it('returns the front element (highest priority) in the queue.', () => {
+  //     const aQueue = new pQueue()
+
+  //     aQueue.enqueue('baz', 300)
+  //     aQueue.enqueue('bar', 200)
+
+  //     expect(() => aQueue.front().to.eql([{name: 'baz', priority: 300}]))
+  //   })
+
+  //   it('returns null if the queue is empty.', () => {
+  //     const aQueue = new pQueue()
+
+  //     expect(() => aQueue.front().to.eql(null))
+  //   })
+  // })
+
+  // context('back()', () => {
+  //   it('returns the back element (lowest priority) in the queue.', () {
+  //     const aQueue = new pQueue()
+
+  //     aQueue.enqueue('bar', 200)
+  //     aQueue.enqueue('foo', 100)
+
+  //     expect(() => aQueue.back().to.eql([{name: 'foo', priority: 100}]))
+  //   })
+
+  //   it('returns null if the queue is empty.', () => {
+  //     const aQueue = new pQueue()
+
+  //     expect(() => pQueue.back().to.eql(null))
+  //   })
+  // })
+
+  context('isEmpty()', () => {
+    it('returns true if the queue is empty.', () => {
+      const aQueue = new pQueue()
+
+      expect(() => aQueue.isEmpty().to.eql(true))
+    })
+
+    it('returns false if queue is false.', () => {
```

Comment:
The test description should clearly reveal the intention of the test.  This description is unclear to me - what does it mean for a queue to be false?  I recommend a description like:
```
returns false if the queue contains no elements
```
This clearly describes the purpose of the `isEmpty` function, and the conditions under which this test should be successful.